modVM/src/org/aion/vm/VirtualMachineManager.java

```diff
@@ -0,0 +1,64 @@
+package org.aion.vm;
+
+import org.aion.vm.VmFactoryImplementation.VM;
+import org.aion.vm.api.interfaces.KernelInterface;
+import org.aion.vm.api.interfaces.VirtualMachine;
+
+/**
+ * A virtual machine factory is a stateful factory that allows a caller to fetch a specified
+ * instance of a {@link VirtualMachine}.
+ *
+ * <p>The order of calls to a VirtualMachineManager must be done in the following order:
+ *
+ * <p>1. initializeAllVirtualMachines 2. getVirtualMachineInstance 3. shutdownAllVirtualMachines
+ *
+ * <p>where the second call can be repeated multiple times before a shutdown.
+ *
+ * <p>This ordering is to ensure that when getVirtualMachineInstance is called, the {@link
+ * VirtualMachine} it returns is initialized and ready to use.
+ */
+public interface VirtualMachineManager {
```

Comment:
IMO, this factory isn't really needed, we can just use its implementation directly (it also depends on an inner class of its implementation VmFactoryImplementation.VM which prevents anyone from using this interface without also using that implementation)

Could you also use the word "Factory" instead of "Manager" in the name (of this interface and/or its impl)

Reply:
I'm going to post an issue about this, it may be a large enough refactoring to warrent more consideration, and I don't want to be changing too much logic at this point.